sapi/fpm/fpm/fpm_unix.c

```diff
@@ -63,6 +64,25 @@ static struct passwd *fpm_unix_get_passwd(struct fpm_worker_pool_s *wp, const ch
 	return pwd;
 }
 
+static inline bool fpm_unix_check_listen_address(struct fpm_worker_pool_s *wp, const char *address, int flags)
+{
+    struct sockaddr_un test_socket;
+    size_t address_length = strlen(address);
+    size_t socket_length = sizeof(test_socket.sun_path);
+
+    if (wp->listen_address_domain != FPM_AF_UNIX) {
+        return true;
+    }
```

Comment:
NIT: you can check this first before getting the length - we no longer require definition at the top (C89) as at least c99 is currently required.

Reply:
Great to hear :) I was worried I have to support some old compiler. Sure will rewrite that